walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,47 +15,66 @@
 package com.google.sps;
 
 import java.lang.Math;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
-import java.util.Comparator;
 import java.util.HashSet;
 import java.util.LinkedList;
 import java.util.List;
-import java.util.ListIterator;
-import java.util.Set;
 
 public final class FindMeetingQuery {
 
   /**
-   * Finds the available {@code TimeRange} for a set of attendees to satisfy a meeting
-   * {@code request}, given some existing {@code events} and their respective attendees.
+   * Finds the available {@code TimeRange} for a set of mandatory and optional attendees to
+   * satisfy a meeting {@code request}, given some existing {@code events} and their respective
+   * attendees. If we can find available {@code TimeRange} for both mandatory and optional
+   * attendees, we return only those {@code TimeRange}. If not, we return all {@code TimeRange}
+   * that satisfy the mandatory attendees. In the special case where there are no mandatory
+   * attendees, we find the {@code TimeRange} that satisfy all optional attendees.
    * The resultant {@code TimeRange} do not overlap with one another and cover all possible
    * empty gaps in the existing {@code events}. These {@code TimeRange} have the same length as
    * or are longer than the requested meeting duration.
    * -- First extracts the existing {@code TimeRange} of those {@code Event} that include any
    * attendees mentioned in the {@code request}. We only need to avoid conflicting with
    * those {@code TimeRange} that involve attendees mentioned in the {@code request}.
-   * -- Then finds the available {@code TimeRange} sandwiched between two already-occupied
-   * {@code TimeRange}.
+   * -- Then finds the available {@code TimeRange} sandwiched between already-occupied
+   * {@code TimeRange}, for mandatory and optional attendees.
    *
    * @param events Existing events that each occupy some {@code TimeRange}.
    * @param request A meeting request for some mandatory and optional attendees and a duration.
    * @return All available {@code TimeRange} that satisfy the meeting request.
    */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
     Collection<String> attendees = request.getAttendees();
```

Comment:
Maybe rename this to mandatoryAttendees because when you do the !attendees.empty() check later on, it gets confusing how this is different than allAttendees

Reply:
Will do, thanks!